### PR TITLE
fixing macos 2022.4 plugin

### PR DESCRIPTION
**Pull request by bam241**
this should allow add the missing lib and to correct its @rpath in the plugin library

### CODE REVIEW

**Comment by bam241** [comment]
this should fix #143 for 2022.4
cc @veeshy

once it is build you should be able to find a new version of the plugin in the build artifacts. 
For your specific version here:
https://github.com/svalinn/Cubit-plugin/actions/runs/3712499162

File: scripts/unix_share_build.sh [hunk 1]
```diff
@@ -74,6 +74,8 @@ function setup_var() {
         CUBIT_PATH="/opt/Coreform-Cubit-2021.3"
     elif [ "$1" == "2021.4" ] ; then
         CUBIT_PATH="/opt/Coreform-Cubit-2021.4"
+    elif [ "$1" == "2022.4" ] ; then
+        CUBIT_PATH="/opt/Coreform-Cubit-2022.4"
```
Comment (bam241) [inline scripts/unix_share_build.sh:78]: this is not used, but I thought it might be better for consistency

**Comment by bam241** [comment]
@veeshy the new build seems to be working for me 

could you confirm ?
```
(Non-Commercial License) Cubit>help export dagmc

Help for words: export & dagmc.

export dagmc <filename> [faceting_tolerance <faceting tolerance>] [length_tolerance <length tolerance>]
     [normal_tolerance <normal tolerance>] [make_watertight] [verbose]
     [fatal_on_curves]


Help for words: export & dagmc.

export dagmc <filename> [faceting_tolerance <faceting tolerance>] [length_tolerance <length tolerance>]
     [normal_tolerance <normal tolerance>] [make_watertight] [verbose]
     [fatal_on_curves]
```

**Comment by veeshy** [comment]
Can confirm it works for me

**Comment by bam241** [comment]
Great.

Could you take a look  @gonuke ?

File: scripts/unix_share_build.sh [hunk 2]
```diff
@@ -289,6 +291,11 @@ function mac_build_plugin_pkg(){
     cp /usr/local/opt/libaec/lib/libsz.dylib .
     install_name_tool -change /usr/local/opt/libaec/lib/libsz.dylib @rpath/libsz.dylib libsvalinn_plugin.so
 
+    if [ "$1" == "2022.4" ] ; then
+    cp /usr/local/opt/libaec/lib/libsz.2.dylib .
+    install_name_tool -change /usr/local/opt/libaec/lib/libsz.2.dylib @rpath/libsz.2.dylib libsvalinn_plugin.so
+    fi
```
Comment (gonuke) [inline scripts/unix_share_build.sh]: What if we made the lib name or version a variable based on the Cubit version, and then use that in various lines that are common (does that make sense?)

File: scripts/unix_share_build.sh (hunk 1 above)
Comment (gonuke) [reply scripts/unix_share_build.sh:78]: Why isn't it used? At least it avoids the exit condition, right?

**Review by gonuke — CHANGES_REQUESTED** [review]
Thanks @bam241 - One small suggestion

File: scripts/unix_share_build.sh (hunk 1 above)
Comment (bam241) [reply scripts/unix_share_build.sh:78]: it is only used if we run this using the build scripts and not the CI.

File: scripts/unix_share_build.sh (hunk 2 above)
Comment (bam241) [reply scripts/unix_share_build.sh]: Do you mean a `$cubit_version` variable instead for using an argument ?

Comment (gonuke) [reply scripts/unix_share_build.sh]: Here's what I was thinking:
```suggestion
    libsz=libsz.dylib
    if [ "$1" == "2022.4" ] ; then
    libsz=libsz.2.dylib
    fi    

    cp /usr/local/opt/libaec/lib/$libsz .
    install_name_tool -change /usr/local/opt/libaec/lib/$libsz @rpath/$libsz libsvalinn_plugin.so
```

**Comment by bam241** [comment]
@gonuke, sorry it took me so long.

next time, if there is just a small suggestion like that, feel free to commit it, and merge it, so this doesn't have to wait for me and my limited availability....

**Comment by bam241** [comment]
ping @gonuke  ?

**Review by gonuke — APPROVED** [review]

**Comment by gonuke** [comment]
Thanks @bam241 - LGTM!

**Pull request merged**